src/walletSdk/Anchor/index.ts

```diff
@@ -195,6 +218,183 @@ export class Anchor {
     }
   }
 
+  /**
+  * Watch all transactions returned from a transfer server. When new or
+  * updated transactions come in, run an `onMessage` callback.
+  *
+  * On initial load, it'll return ALL pending transactions via onMessage.
+  * Subsequent messages will be any one of these events:
+  *  * Any new transaction appears
+  *  * Any of the initial pending transactions change any state
+  *
+  * You may also provide an array of transaction ids, `watchlist`, and this
+  * watcher will always react to transactions whose ids are in the watchlist.
+  */
+  watchAllTransactions(params: {
+    authToken: string;
+    assetCode: string;
+    onMessage: (transaction) => void;
+    onError: (error: any) => void;
+    watchlist?: string[];
+    timeout?: number;
+    isRetry?: boolean;
+    noOlderThan?: string;
+    kind?: string;
+    lang?: string;
+  }): WatcherResponse {
+    const {
+      authToken,
+      assetCode,
+      onMessage,
+      onError,
+      watchlist = [],
+      timeout = 5000,
+      isRetry = false,
+      ...otherParams
+    } = params;
+    
+    // make an object map out of watchlist
+    const watchlistMap: any = watchlist.reduce(
```

Comment:
nit: I think we can remove the `any` here and have typescript infer it?
```suggestion
    const watchlistMap = watchlist.reduce(
```